addons/skin.estuary/xml/Variables.xml

```diff
@@ -318,11 +318,12 @@
 		<value>$INFO[ListItem.PictureDateTime]</value>
 	</variable>
 	<variable name="NowPlayingBreadcrumbsVar">
-  		<value condition="MusicPlayer.Content(livetv) + Player.HasAudio">$INFO[MusicPlayer.Title]</value> 
+		<value condition="MusicPlayer.Content(livetv) + Player.HasAudio + !String.IsEmpty(MusicPlayer.Artist) + !String.IsEmpty(MusicPlayer.Title)">$INFO[MusicPlayer.Artist,, - ]$INFO[MusicPlayer.Title]</value>
```

Comment:
what should be displayed in case either Artist or Title (or both) return empty?
is it even possible that `MusicPlayer.Title` ever returns empty?

i think you'll need 2 conditions here:
```
<value condition="MusicPlayer.Content(livetv) + Player.HasAudio + !String.IsEmpty(MusicPlayer.Artist)">$INFO[MusicPlayer.Artist,, - ]$INFO[MusicPlayer.Title]</value>
<value condition="MusicPlayer.Content(livetv) + Player.HasAudio">$INFO[MusicPlayer.Title]</value>
```

Reply:
For Content(LiveTV):

Yes, Musicplayer.Title might return empty, for instance if a shoutcast stream has no (song) title or the PVR channel does have no EPG data (and setting hide 'no info avail' is set to true).

If both values are empty, it is important that we we fall thru until the stationname check, which might evaluate to non-empty for shoutcast. Last resort is "Now playing..."

So, the conditions you propose will unfortunately prevent the fall-thru to stationname check, and now playing

Reply:
ok, next try:
```
<value condition="MusicPlayer.Content(livetv) + Player.HasAudio + !String.IsEmpty(MusicPlayer.Artist) + !String.IsEmpty(MusicPlayer.Title)">$INFO[MusicPlayer.Artist,, - ]$INFO[MusicPlayer.Title]</value>
<value condition="MusicPlayer.Content(livetv) + Player.HasAudio + [!String.IsEmpty(MusicPlayer.Artist) | !String.IsEmpty(MusicPlayer.Title)]">$INFO[MusicPlayer.Artist]$INFO[MusicPlayer.Title]</value>
```

first condition will match if both Artist **and** Title are set
second matches if Artist **or** Title is set
we fall through of both return empty.

Reply:
@ronie I think we can just revert the change to that line. Makes no sense.

Reply:
Yep, what we need is only one line as first condition

`<value condition="MusicPlayer.Content(livetv) + Player.HasAudio">$INFO[MusicPlayer.Artist,, - ]$INFO[MusicPlayer.Title]</value>`

MusicPlayer.Title can't be empty.

Fall-thru to MusicPlayer.StationName makes no sense for PVR as OSD sublabel will contain the channel name, which most probably is the same as station name. If not, doesn't matter. One name for the baby on the upper left is sufficient, channel name wins, then.

Reply:
Changed. Force-pushed. @ronie please re-review. Thanks.